### DISCUSSION TITLE
possible tutorial issue with arrow middleware

### DISCUSSION THREAD

**Discussion by mkelandis** [floating-ui/floating-ui#2026] [Q&A] [answered]
In the section of the tutorial describing the arrow middleware, it appears that the x and y coordinates are being set using the type "arrowX" instead of the value x...

![image](https://user-images.githubusercontent.com/2457369/206756852-0bba451f-2202-460b-a3c0-8ddd756b27c2.png)

It seems like x and y are already defined in the Promise returned by computePosition above this code, so I'm guessing these values should be renamed. In my code I did this instead:

```ts
    // Accessing the data
    const arrowCoords = middlewareData.arrow;

    const staticSide = {
      top: 'bottom',
      right: 'left',
      bottom: 'top',
      left: 'right'
    }[placement.split('-')[0]];

    Object.assign(arrowElement.style, {
      left: arrowCoords.x != null ? `${arrowCoords.x}px` : '',
      top: arrowCoords.y != null ? `${arrowCoords.y}px` : '',
      right: '',
      bottom: '',
      [staticSide]: '-4px'
    });

```

**Comment by atomiks** [comment, marked as answer]
That's what `{x: arrowX y: arrowY}` is doing

**Comment by mkelandis** [comment]
forgive me - That's the first time I've ever seen the right hand side of json props used in a deconstructing assignment like that. it did indeed work just threw my brain into a loop (for the entire internet to witness!).
